src/request.ts

```diff
@@ -339,20 +339,14 @@ class DatastoreRequest {
    *   const apiResponse = data[0];
    * });
    */
-  delete(keys: Entities):
-      void|Promise<google.datastore.v1.Datastore.CommitCallback>;
-  delete(
-      keys: Entities,
-      callback: google.datastore.v1.Datastore.CommitCallback): void;
-  delete(
-      keys: Entities, gaxOptions: CallOptions,
-      callback: google.datastore.v1.Datastore.CommitCallback): void;
+  delete(): Promise<google.datastore.v1.CommitResponse>;
```

Comment:
That isn't the right type.  All of the promises for our APIs resolve with `Promise<[SomeType, OriginalProtoBuf]>` or something like that.  This should probably be `Promise<[google.datastore.v1.CommitResponse]>`.  That thing is long enough that I would create:
```js
export type CommitResponse = [google.datastore.v1.Datastore.CommitCallback];
```

And then just use that in the other signatures.

Reply:
Yikes, yeah I kind of knew this wouldn't fly, was trying to be space-efficient. Fixed  👍